app/Services/AuthService.php

```diff
@@ -27,70 +31,234 @@ public function __construct(User $userObj)
 
     public function signup($inputs)
     {
-        $user = $this->userObj->create(Arr::only($inputs, ['name', 'email', 'password']));
+        $user = $this->userObj->create($inputs);
+        $otp = mt_rand(100000, 999999);
+        $this->userOtpService->store(['otp' => $otp, 'user_id' => $user->id, 'otp_for' => 'verification']);
 
-        return $user;
+        try {
+            VerifyUserMail::dispatch($user, $otp);
+        } catch (\Exception $e) {
+            Log::info('User verification mail failed.' . $e->getMessage());
+        }
+
+        $data = [
+            'status' => true,
+            'message' => __('message.userSignUpSuccess'),
+            'user' => new UserResource($user),
+            'token' => $user->createToken(config('app.name'))->plainTextToken,
+        ];
+
+        return $data;
+    }
+
+    public function verifyOtp($inputs)
+    {
+        $user = $this->userObj->whereEmail($inputs['email'])->first();
+
+        if (empty($user)) {
+            $data = [
+                'status' => false,
+                'message' =>  __('message.emailNotExist')
+            ];
+            return $data;
+        }
+
+        $userOtp = $this->userOtpObj->whereUserId($user['id'])->whereOtp($inputs['otp'])->where('otp_for', 'verification')->first();
+
+        if (empty($userOtp)) {
+            $data = [
+                'status' => false,
+                'message' =>  __('message.invalidOtp')
+            ];
+            return $data;
+        }
+
+        $expirationTime = config('site.otpExpirationTimeInMinutes');
+        $expirationDate = Carbon::parse($userOtp['created_at'])->addMinutes($expirationTime)->format('Y-m-d H:i:s');
```

Comment:
Can we move this functionality to `isOtpExpired($created_at`) on `UserOtpService`? 

So that in future if we want to verify the OTP expiration for another module we don't need to add the same code.